### PR TITLE
site: remove useless any

### PR DESCRIPTION
**Pull request by thinkasany**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution
1.child类型是可以用any的，不需要断言
image

image

2. treeIcon 是否展示 TreeNode title 前的图标，仅仅为boolean
image
image

3.
似乎缺少一个像tree组件那样子使用showicon的demo
image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     type: remove useless any      |
| 🇨🇳 Chinese |      删除无用的any类型     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 3f7a8b4</samp>

This pull request enhances the tree select component and its demo by adding custom icons and fixing a prop type issue. It also improves the `useMenu` hook by handling undefined values more gracefully.

### <samp>🤖 Generated by Copilot at 3f7a8b4</samp>

*  Import and use `CarryOutOutlined` icon for custom tree nodes in demo ([link](https://github.com/ant-design/ant-design/pull/44018/files?diff=unified&w=0#diff-2fc074e66eb294019dc396174b3e8b3f9d041d6f0a0c67545403994ed3b2cbbeR2), [link](https://github.com/ant-design/ant-design/pull/44018/files?diff=unified&w=0#diff-2fc074e66eb294019dc396174b3e8b3f9d041d6f0a0c67545403994ed3b2cbbeL8-R24), [link](https://github.com/ant-design/ant-design/pull/44018/files?diff=unified&w=0#diff-2fc074e66eb294019dc396174b3e8b3f9d041d6f0a0c67545403994ed3b2cbbeL26-R36))
*  Add `showIcon` state and switch component to toggle custom icons in demo ([link](https://github.com/ant-design/ant-design/pull/44018/files?diff=unified&w=0#diff-2fc074e66eb294019dc396174b3e8b3f9d041d6f0a0c67545403994ed3b2cbbeL40-R57), [link](https://github.com/ant-design/ant-design/pull/44018/files?diff=unified&w=0#diff-2fc074e66eb294019dc396174b3e8b3f9d041d6f0a0c67545403994ed3b2cbbeR74))
*  Cast `treeIcon` prop to boolean in `TreeSelect` component ([link](https://github.com/ant-design/ant-design/pull/44018/files?diff=unified&w=0#diff-f4ebf787ec0bc13112f49b0d9a2df0bfa73f8ea80cec9a6c92d311e120fa6ffeL277-R277))
*  Use optional chaining to access `frontmatter` property of possible undefined objects in `useMenu` hook ([link](https://github.com/ant-design/ant-design/pull/44018/files?diff=unified&w=0#diff-97b7cacb6d4bd1445cc4d73ae072d3aac76d65d4c10eb29328c6cc8aef1d6649L55-R55), [link](https://github.com/ant-design/ant-design/pull/44018/files?diff=unified&w=0#diff-97b7cacb6d4bd1445cc4d73ae072d3aac76d65d4c10eb29328c6cc8aef1d6649L110-R114))

### CODE REVIEW

File: components/tree-select/demo/treeLine.tsx [hunk 1]
```diff
@@ -1,32 +1,39 @@
 import React, { useState } from 'react';
+import { CarryOutOutlined } from '@ant-design/icons';
```
Comment (li-jia-nan) [inline components/tree-select/demo/treeLine.tsx:2]: 改 demo 和改 type 拆开，一个 PR 只做一件事情，保证 PR 的纯粹性

File: components/tree-select/index.tsx [hunk 2]
```diff
@@ -274,7 +274,7 @@ const InternalTreeSelect = <
       removeIcon={removeIcon}
       allowClear={mergedAllowClear}
       switcherIcon={renderSwitcherIcon}
-      showTreeIcon={treeIcon as any}
+      showTreeIcon={!!treeIcon}
```
Comment (li-jia-nan) [inline components/tree-select/index.tsx:277]: 这样不等价了，这里没问题就暂时不用改

Comment (thinkasany) [reply components/tree-select/index.tsx:277]: sorry，问一下不等价的情况是啥，不应该是要么undefined，要么true false, 这三种么
image

Comment (li-jia-nan) [reply components/tree-select/index.tsx:277]: 你不能保证底层 `rc-tree-select` 内部对 showTreeIcon 进行了 to boolean 的操作，如果没有的话，那这里就不等价了

Comment (li-jia-nan) [reply components/tree-select/index.tsx:277]: 文档应该根据实现产出的，而不是实现根着文档写，文档只是暴露给用户看的，不需要考虑实现

File: components/tree-select/demo/treeLine.tsx (hunk 1 above)
Comment (thinkasany) [reply components/tree-select/demo/treeLine.tsx:2]: 好的， 已拆出去了，https://github.com/ant-design/ant-design/pull/44019

File: components/tree-select/index.tsx (hunk 2 above)
Comment (thinkasany) [reply components/tree-select/index.tsx:277]: 好的 感谢解答，我去看看rc-tree-select的逻辑

**Review by li-jia-nan — APPROVED** [review]

**Pull request merged**